File: myFinance_improved_V2/lib/features/inventory_analysis/discrepancy/domain/repositories/discrepancy_repository.dart
import 'package:dartz/dartz.dart';

import '../../../../../core/errors/failures.dart';
import '../entities/discrepancy_overview.dart';

/// Discrepancy Analytics Repository Interface
///
/// Clean Architecture's Domain Layer interface for
/// inventory discrepancy analysis data retrieval.
abstract class DiscrepancyRepository {
  /// Get discrepancy overview data
  /// RPC: get_discrepancy_overview
  ///
  /// [companyId] Company ID (required)
  /// [period] Period filter ('7d', '30d', 'all') (optional)
  Future<Either<Failure, DiscrepancyOverview>> getDiscrepancyOverview({
    required String companyId,
    String? period,
  });
}

File: website/src/features/inventory-analysis/data/repositories/DiscrepancyRepository.ts
/**
 * Discrepancy Repository
 * Repository layer for discrepancy overview data
 */

import {
  discrepancyDataSource,
  GetDiscrepancyOverviewParams,
} from '../datasources/DiscrepancyDataSource';
import type { DiscrepancyOverview } from '../../domain/entities/discrepancyOverview';

export class DiscrepancyRepository {
  async getDiscrepancyOverview(params: GetDiscrepancyOverviewParams): Promise<DiscrepancyOverview> {
    return discrepancyDataSource.getDiscrepancyOverview(params);
  }
}

// Singleton instance
export const discrepancyRepository = new DiscrepancyRepository();
